src/stores/WalletStore.ts

```diff
@@ -102,16 +116,45 @@ export default class WalletStore {
 
   @action
   public logout = () => {
-    this.app.walletStore.stopPolling();
+    this.reset();
     this.app.routerStore.push('/login');
   }
 
+  @action
+  public reset = () => {
+    this.stopPolling();
+    this.info =  INIT_VALUES.info;
+    this.loggedInAccount = INIT_VALUES.loggedInAccount;
+    this.wallet = INIT_VALUES.wallet;
+  }
+
+  @action
+  public resetWithNetwork = () => {
+    this.reset();
+    this.accounts = INIT_VALUES.accounts;
+  }
+
   @action
   private recoverWallet(mnemonic: string) {
-    const network = networks.testnet;
+    const network = this.app.networkStore.network;
     this.wallet = network.fromMnemonic(mnemonic);
   }
 
+  private handleGetChromeStorageReturn(storageAccounts: any[]) {
```

Comment:
the type should be `Account[]`

Reply:
also, believe this should be an action since you are changing store values. let's use fat arrow functions for all functions as well. `const funcName = () => { }`. its the newest syntax and it auto-binds for you.

Reply:
nit: `routeToInitialPage()`